models/Element/AbstractElement.php

```diff
@@ -222,20 +224,27 @@ public function isAllowed($type, ?Model\User\AbstractUser $user = null)
         }
 
         //everything is allowed for admin
+        /** @var Model\User $user */
         if ($user->isAdmin()) {
             return true;
         }
 
-        return $this->getDao()->isAllowed($type, $user);
+        /** @var Model\Document\Dao|Model\Asset|Dao|Model\DataObject\AbstractObject\Dao $dao */
+        $dao = $this->getDao();
+        return $dao->isAllowed($type, $user);
     }
 
     public function unlockPropagate()
     {
         $type = Service::getType($this);
-        $ids = $this->getDao()->unlockPropagate();
+
+        /** @var Model\Document\Dao|Model\Asset|Dao|Model\DataObject\AbstractObject\Dao $dao */
+        $dao = $this->getDao();
+        $ids = $dao->unlockPropagate();
 
         // invalidate cache items
         foreach ($ids as $id) {
+            /** @var Model\Document\Dao|Model\Asset|Dao|Model\DataObject\AbstractObject\Dao $element */
```

Comment:
The Dao classes haven't the clearDependentCache method, right? We should add the clearDependentCache method to the ElementInterface.

Reply:
Mhm, ElementInterface or AbstractElement ?
Then I would say we should add things like setId() as well

Reply:
I would prefer the interface. It should have alle necessary methods. The `Service::getElementById()` function have as return also the interface.